catalog/policies/git_push.tracked-run.rego

```diff
@@ -0,0 +1,96 @@
+# https://docs.spacelift.io/concepts/policy/git-push-policy
+# GIT_PUSH policy that triggers tracked runs when component files or YAML config files are modified on commits to the default branch
+
+package spacelift
+
+# Get all affected files
+# `input.push.affected_files` contains a list of file names (relative to the project root)
+# that were changed in the current push to the default branch
+affected_files := input.push.affected_files
+
+# Track these extensions in the project folder
+tracked_extensions := {".tf", ".tf.json", ".tfvars", ".yaml", ".yml", ".tpl", ".sh", ".shell", ".bash", ".json"}
+
+# Project root
+project_root := input.stack.project_root
+
+# Ignore if any of the `ignore` rules evaluates to `true`
+ignore  {
+    not project_affected
+    not stack_config_affected
+}
+
+# If pre-commit hooks make changes, they are not semantic changes and can and should be ignored
+ignore  {
+    input.push.message == "pre-commit fixes"
+}
+
+# Track if project files are affected and the push was to the stack's tracked branch
+# https://docs.spacelift.io/concepts/run/tracked
+track {
+    project_affected
+    input.push.branch == input.stack.branch
+}
+
+# Track if stack config files are affected and the push was to the stacks' tracked branch
+# https://docs.spacelift.io/concepts/run/tracked
+track {
+    stack_config_affected
+    input.push.branch == input.stack.branch
+}
+
+# Check if any of the tracked extensions were modified in the project folder
+# https://www.openpolicyagent.org/docs/latest/policy-language/#some-keyword
+# https://www.openpolicyagent.org/docs/latest/policy-language/#variable-keys
+# https://www.openpolicyagent.org/docs/latest/policy-reference/#iteration
+project_affected {
+    some i, j
+    startswith(affected_files[i], project_root)
+    endswith(affected_files[i], tracked_extensions[j])
+}
+
+# Split the stack name into a list
+stack_name_parts := split(input.stack.name, "-")
+
+# Check if the environment has been modified
+stack_config_affected {
+    contains(affected_files[_], concat("-", [stack_name_parts[0], stack_name_parts[1]]))
```

Comment:
if the name of the stack is `v1-datadog.yaml`, would this cause issues here ?

Reply:
it will be caught by the main code that detects the `deps:` labels.
It's here for historical reasons, I did not remove it to not make too many changes at once.